test/integration/backward_compatible/parallel/map/NearCachedMapTest.js

```diff
@@ -102,7 +102,7 @@ describe('NearCachedMapTest', function () {
                 }
                 await map1.get('key1');
                 await map2.remove('key1');
-                const val = await TestUtil.promiseLater(1000, map1.get.bind(map1, 'key1'));
+                const val = await TestUtil.promiseLater(5000, map1.get.bind(map1, 'key1'));
                 expectStats(map1, 0, 2, 1);
```

Comment:
just to be sure I understand correctly: `map1` puts a value into the map, `map2` removes that value, and it's therefore expected that `map1` NearCache gets notified and stops returning the value. Method `promiseLater` is going to wait for the specified time, *then* get the value. If the time value is too small, the test fail because you *still* get the value, most probably because the value has not been removed *yet*. Correct?

We have the same issues in .NET and at times some tests fail because the GitHub runner seems to pause for 5-10 seconds. I would *really* like to use mock servers for this, so we can control timeouts without depending on actual servers.

Reply:
> just to be sure I understand correctly: map1 puts a value into the map, map2 removes that value, and it's therefore expected that map1 NearCache gets notified and stops returning the value. Method promiseLater is going to wait for the specified time, then get the value. If the time value is too small, the test fail because you still get the value, most probably because the value has not been removed yet. Correct?

exactly.

> We have the same issues in .NET and at times some tests fail because the GitHub runner seems to pause for 5-10 seconds. I would really like to use mock servers for this, so we can control timeouts without depending on actual servers.

I can't imagine how that mocking would work but I like the idea of solving these failures.